template/registry/number.go

```diff
@@ -0,0 +1,240 @@
+				Description:    `When to display the sign for the number. "negative" value is Experimental.`,
+				PossibleValues: []any{"auto", "always", "exceptZero", "negative", "never"},
+				Default:        "auto",
+			},
+			{
+				Name:           "style",
+				Description:    "The formatting style to use.",
+				PossibleValues: []any{"decimal", "currency", "percent", "unit"},
+				Default:        "decimal",
+			},
+			{
+				Name: "unit",
+				Description: `The unit to use in unit formatting.
+Possible values are core unit identifiers, defined in UTS #35, Part 2, Section 6.
+A subset of units from the full list was selected for use in ECMAScript.
+Pairs of simple units can be concatenated with "-per-" to make a compound unit.
+There is no default value; if the style is "unit", the unit property must be provided.`,
+				ValidateValue: isPositiveInteger,
+			},
+			{
+				Name:           "unitDisplay",
+				Description:    "The unit formatting style to use in unit formatting.",
+				PossibleValues: []any{"long", "short", "narrow"},
+				Default:        "short",
+			},
+			{
+				Name: "minimumIntegerDigits",
+				Description: `The minimum number of integer digits to use.
+A value with a smaller number of integer digits than this number will be
+left-padded with zeros (to the specified length) when formatted.`,
+				ValidateValue: isPositiveInteger,
+				Default:       1,
+			},
+			{
+				Name: "minimumFractionDigits",
+				Description: `The minimum number of fraction digits to use.
+The default for plain number and percent formatting is 0;
+the default for currency formatting is the number of minor unit digits provided by
+the ISO 4217 currency code list (2 if the list doesn't provide that information).`,
+				ValidateValue: isPositiveInteger,
+			},
+			{
+				Name: "maximumFractionDigits",
+				Description: `The maximum number of fraction digits to use.
+The default for plain number formatting is the larger of minimumFractionDigits and 3;
+the default for currency formatting is the larger of minimumFractionDigits and the number of
+minor
+unit digits provided by the ISO 4217 currency code list (2 if the list doesn't provide that
+information);
+the default for percent formatting is the larger of minimumFractionDigits and 0.`,
+				ValidateValue: isPositiveInteger,
+			},
+			{
+				Name:          "minimumSignificantDigits",
+				Description:   `The minimum number of significant digits to use.`,
+				ValidateValue: isPositiveInteger,
+				Default:       1,
+			},
+			{
+				Name:          "maximumSignificantDigits",
+				Description:   `The maximum number of significant digits to use.`,
+				ValidateValue: isPositiveInteger,
+				Default:       21, //nolint:gomnd
+			},
+		},
+	},
+}
+
+// TODO: supports only style and signDisplay options.
+func numberF(input any, options map[string]any) (any, error) {
+	num, err := castAs[float64](input)
+	if err != nil {
+		return nil, fmt.Errorf("convert input to float64: %w", err)
+	}
+
+	if options == nil {
```

Comment:
Check `len` instead?

```suggestion
	if len(options) == 0 {
```